WordPress/Sniffs/WP/I18nSniff.php

```diff
@@ -341,15 +372,32 @@ protected function check_argument_tokens( $context ) {
 		$tokens    = $context['tokens'];
 		$arg_name  = $context['arg_name'];
 		$is_error  = empty( $context['warning'] );
-		$content   = $tokens[0]['content'];
+		$content   = isset( $tokens[0] ) ? $tokens[0]['content'] : '';
 
 		if ( empty( $tokens ) || 0 === count( $tokens ) ) {
 			$code = $this->string_to_errorcode( 'MissingArg' . ucfirst( $arg_name ) );
-			if ( 'domain' !== $arg_name || ( ! empty( $this->text_domain ) && ! in_array( 'default', $this->text_domain, true ) ) ) {
+			if ( 'domain' !== $arg_name ) {
+				$this->addMessage( 'Missing $%s arg.', $stack_ptr, $is_error, $code, array( $arg_name ) );
+				return false;
+			}
+
+			// Ok, we're examining a text domain, now deal correctly with the 'default' text domain.
+			if ( true === $this->text_domain_is_default ) {
```

Comment:
This could be a separate `if` (and the next `elseif` becomes an `if`) since we return here.

Reply:
Done.